issue/4478-payment-tutorial-tablets

nbradbury opened the pull request:
Fixes #4478 - previously the payment tutorial was visually broken on tablets. This PR corrects this. Tablet portrait and landscape shots below. To test, I recommend changing the [onboarding click listener ](https://github.com/woocommerce/woocommerce-android/blob/3f310e4e6bb0d27c0c1d27f2bd5957c3d230bb98/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt#L124)to this, so you can easily view the tutorial:

```
childFragmentManager.beginTransaction()
                .add(CardReaderTutorialDialogFragment(), "TAG")
                .commitAllowingStateLoss()
```

![landscape](https://user-images.githubusercontent.com/3903757/126791085-77841b24-6f42-4a96-bde3-0a3684de82bf.png)
![portrait](https://user-images.githubusercontent.com/3903757/126791089-035d92b3-4087-4d95-b04f-b3a77fa5ca8a.png)





Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

nbradbury committed:
Fixed tutorial on tablets

malinajirka commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/tutorial/CardReaderTutorialViewPagerItemFragment.kt:
```diff
@@ -25,22 +25,24 @@ class CardReaderTutorialViewPagerItemFragment : Fragment(R.layout.fragment_card_
-            } else {
-                (DisplayUtils.getDisplayPixelWidth() * RATIO_PORTRAIT).toInt()
+            // adjust the view sizes based on orientation (skipped for tablets)
+            if (!isTablet) {
```
❓ I'm not familiar with the code - I realize it wasn't added in this PR, but I'd like to understand it since this PR touches the logic. What is the purpose of manually changing view dimentions? Can't we leave it up to the OS to switch between landscape and portrait layouts?

malinajirka reviewed (commented):
Thanks @nbradbury for the quick fix!

I noticed that the layout doesn't have any padding on the sides on tablets, any idea why?
<img src="https://user-images.githubusercontent.com/2261188/126794618-93415adf-e411-49eb-b080-2dfcba2633c1.png" width="250px" />
<img src="https://user-images.githubusercontent.com/2261188/126794627-19067714-0714-4af0-adf5-2e9125c87c56.png" width="250px" />

nbradbury committed:
Don't adjust view ratios

nbradbury committed:
Added vertical padding to ViewPager item layout

nbradbury commented:
> What is the purpose of manually changing view dimentions? 

@malinajirka You're right, that's not necessary. Addressed in afe6a60.

> I noticed that the layout doesn't have any padding on the sides on tablets, any idea why?

Strange, it looks fine on my Nexus 10 emulator. I added some padding to the ViewPager item in afb73f1 and hopefully that fixes it for you.

malinajirka reviewed (approved):
I've tested this on phone and tablet in both landscape and portrait and it all looks great, thanks for working on this! ;)
